docs/usage/configuration/config_documentation.md

```diff
@@ -3053,8 +3053,12 @@ Options for each entry include:
    values are `client_secret_basic` (default), `client_secret_post` and
    `none`.
 
+* `code_challenge_method`: proof key for code exchange method used when requesting
+  and exchanging the token. Valid values are `null` (default, disables support),
+  `plain`, and `S256`.
```

Comment:
I don't have a strong opinion on this, but I would probably enable it by default if the server supports it. Something like `use_pkce: true|false|auto`

Reply:
Default to `auto`, but `true` would mean "use `S256`"? That could work. Not sure what we would do if another algorithm was added, but 🤷

Reply:
> `true` would mean "use `S256`"

Yep, for cases where don't have OIDC discovery and thus don't know if the server supports it

Reply:
I did this in `5e910b8d76e539fb9e2c02799cd9c258395f9c96, I ended up using `pkce_method: auto|always`, which seems to match our other options better. (Hopefully we don't need to disable it ever...) I also didn't want to use `true`/`false` since YAML coerces those to booleans.